src/main/java/com/exactpro/th2/common/module/FileConfigurationProviderFactoryShim.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.module;
+
+import com.exactpro.th2.common.ConfigurationProviderFactory;
+import com.exactpro.th2.common.module.provider.FileConfigurationProviderFactory;
+import com.google.auto.service.AutoService;
+import kotlin.Deprecated;
+import kotlin.DeprecationLevel;
+
+@AutoService(ConfigurationProviderFactory.class)
+@Deprecated(
+        message = "Don't use this class, it is only meant to be captured by ServiceLoader",
+        level = DeprecationLevel.HIDDEN)
+public final class FileConfigurationProviderFactoryShim extends FileConfigurationProviderFactory {
```

Comment:
What is it for?

Reply:
AutoService annotation just doesn't work with kotlin classes, it doesn't see them. So, you need to either write in Java, or use such shims.

Reply:
It does not work with Kotlin classes because you need to add kapt plugin
See the example here:
https://github.com/th2-net/th2-data-processor-zephyr/blob/master/build.gradle#L42